x-pack/plugins/discover_log_explorer/public/components/dataset_selector/dataset_selector.tsx

```diff
@@ -22,20 +22,24 @@ import { DatasetsPopover } from './sub_components/datasets_popover';
 import { DatasetSkeleton } from './sub_components/datasets_skeleton';
 import { SearchControls } from './sub_components/search_controls';
 import { DatasetSelectorProps } from './types';
-import { buildIntegrationsTree } from './utils';
+import {
+  buildIntegrationsTree,
+  createAllLogDatasetsItem,
+  createUnmanagedDatasetsItem,
+  createIntegrationStatusItem,
+} from './utils';
 
 /**
  * Lazy load hidden components
  */
 const DatasetsList = dynamic(() => import('./sub_components/datasets_list'), {
   fallback: <DatasetSkeleton />,
 });
-const IntegrationsListStatus = dynamic(() => import('./sub_components/integrations_list_status'));
 
 export function DatasetSelector({
   datasets,
   datasetsError,
-  initialSelected,
+  datasetSelection,
```

Comment:
The component relies on this controlled property to present what the current selection is, while the state machine will notify with a new selection instance anytime the user clicks on an entry.